test/accelerators/NNPA/numerical/TestGRU-NNPA.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <rapidcheck.h>
+
+#include "llvm/Support/FileSystem.h"
+
+#include "include/OnnxMlirRuntime.h"
+#include "src/Runtime/OMTensorHelper.hpp"
+#include "test/modellib/ModelLib.hpp"
+
+static const llvm::StringRef SHARED_LIB_BASE("./TestGRUNNPA_main_graph");
+
+using namespace mlir;
+
+namespace onnx_mlir {
+namespace test {
+
+// Returns whether onnx-mlir compiled GRU is producing the same results as a
+// naive implementation of GRU for a specific set of GRU
+// parameters/configuration.
+bool isOMGRUTheSameAsNaiveImplFor(const int direction, const int S, const int B,
+    const int I, const int H, const int linearBeforeReset,
+    bool isDynamicS = false, bool isDynamicB = false) {
+
+  GRULibBuilder gru(SHARED_LIB_BASE.str(), direction, S, B, I, H,
+      linearBeforeReset, isDynamicS, isDynamicB);
+  return gru.build() && gru.compileAndLoad() && gru.prepareInputs() &&
+         gru.run() && gru.verifyOutputs();
```

Comment:
Do we maybe need to add a RTOL (& ATOL) parameter to verifyOutputs() because I would expect NNPA errors to be larger than CPUs? If so, I would have a default value of the traditional RTOL and then a specific value. You would then need to change the `areCloseFloat` as well to take a RTOL and ATOL parameter

Reply:
Yes. We need to use different RTOL and ATOL. They can be configured by using environment variables `TEST_ATOL` and `TEST_RTOL`. So, I set them in `test/accelerators/NNPA/numerical/CMakeLists.txt` as follows.

```
set_tests_properties(${test_name} PROPERTIES ENVIRONMENT "TEST_ATOL=0.01;TEST_RTOL=0.05};LD_PRELOAD=${ZDNN_LIB}")
```